### ISSUE TITLE
Not parsing string function $contains as per JSONata

### ISSUE THREAD

**Issue by cdicsu** [IBM/JSONata4Java#17]
Using the JSON body: `{
    "response": {
        "addressLine1": "1111 AVENUE O",
        "city": "GALVESTON",
        "state": "TX",
        "zip": "77550",
        "zip4": "8345",
        "validationMessages": [{
            "code": "HR",
            "message": "THE ADDRESS IS A HIGH RISE.",
            "severity": "I"
        }]
    },
    "conversationId": "794c81c9-aad9-44d9-8add-6152b3174606",
    "transactionId": "f115f062-ca77-45e0-a9e1-3a1ecefc38ca"
}`

and the JSONata expression: `true in response.validationMessages.$contains(severity, 'I')`
returns true in the JSONata Exerciser, but throws `com.api.jsonata4java.expressions.EvaluateException: Context value is not a compatible type with argument 1 of function "$contains"` in JSONata4Java.

For this package, in order to get the expression to work properly, I had to change it to `"$contains(validationMessages.severity, 'I')"` and then while running this same expression in the JSONata Exerciser, I get the same error response as above.  In addition to this, when the expression that works with JSONata4Java for the validateMessages list is used on the same list with more than one element, the same exception mentioned above is thrown.  It only works on the list consisting of one element.  Using JSONata Exerciser, `true in response.validationMessages.$contains(severity, 'I')` works on a list consisting of n elements.

TL;DR What works in JSONata Exerciser doesn't work in JSONata4Java, and what works in JSONata4Java doesn't work in JSONata Exerciser, pertaining to the string function $contains().  TIA for any insight you may be able to provide.

**Comment by wnm3** [comment]
Thank you for the detailed issue. I had taken the documentation [here](https://docs.jsonata.org/string-functions#contains) too literally so was only allowing a String as input. I'll try to revise to function more like the JavaScript version to search keys in objects, etc.

**Comment by wnm3** [comment]
I'll check in my updates to ContainsFunction. My testing is shown below:

Attempting to load JSON from file: responses.json
Using json:
{
  "response" : {
    "addressLine1" : "1111 AVENUE O",
    "city" : "GALVESTON",
    "state" : "TX",
    "zip" : "77550",
    "zip4" : "8345",
    "validationMessages" : [ {
      "code" : "HR",
      "message" : "THE ADDRESS IS A HIGH RISE.",
      "severity" : "I"
    }, {
      "code" : "HR",
      "message" : "THE ADDRESS IS A HIGH RISE.",
      "severity" : "J"
    } ]
  },
  "conversationId" : "794c81c9-aad9-44d9-8add-6152b3174606",
  "transactionId" : "f115f062-ca77-45e0-a9e1-3a1ecefc38ca"
}
Enter jsonata expression (or q to quit):
response.validationMessages.$contains(severity, 'I')
[ true, false ]
Enter jsonata expression (or q to quit):
response.validationMessages.$contains(severity, 'J')
[ false, true ]
Enter jsonata expression (or q to quit):
response.validationMessages.$contains(severity, 'X')
[ false, false ]
Enter jsonata expression (or q to quit):
response.validationMessages.$contains("severity", 'I')
[ false, false ]
Enter jsonata expression (or q to quit):
response.validationMessages.$contains(severity, 'i')
[ false, false ]
Enter jsonata expression (or q to quit):
response.validationMessages.$contains("severity", 'i')
[ true, true ]
Enter jsonata expression (or q to quit):
$contains("severity","i")
true
Enter jsonata expression (or q to quit):
$contains("severity","I")
false
Enter jsonata expression (or q to quit):
$contains(severity,"i")
** no match **
Enter jsonata expression (or q to quit):
q
Goodbye

**Comment by cdicsu** [comment]
Great!  Thanks so much!  I'll follow here for any further updates!

**Comment by wnm3** [comment]
Posted updates -- will try to work on the other issue later today. Once #18 is resolved, I'll push to Maven Central as v1.0.9

**Comment by cdicsu** [comment]
Thanks!  pulled new version and $contains looks like it's working well!
